docs/data/sdks/javascript/index.md

```diff
@@ -983,23 +983,86 @@ var exitCallback = function {
 amplitude.getInstance().init('API_KEY', 'USER_ID', { onExitPage: exitCallback });
 ```
 
-### Custom device IDs
+### Device ID lifecycle
 
-By default, device IDs are randomly generated UUIDs. You can define a custom device ID by setting it as a configuration option or by calling `setDeviceId`.
+By default Amplitude Javascript SDK uses a base64id as device id in order to have more compact string IDs. A 22-character base64id has the same range as a 128-bit UUID, which is represented as a 36 character string.
+
+On instance initialization, the SDK initializes the device ID by the following order:
+
+1. Device id in configuration on initialization
+2. Device id from url param if `configuration.deviceIdFromUrlParam` is true
+3. Device id in cookie storage
+
+!!!Note
+    If `deviceIdFromUrlParam` is `true`, then the SDK will parse the device ID value from the URL parameter "amp_device_id" if available. For example, device ID “123456789” is passed in from http://example.com/?amp_device_id=123456789
+
+The SDK stores device ID in cookie by default. The name of the cookie is composed of a "amp_" prefix and the first six digits of the API key. So if a user clears cookies, uses another device or uses browser privacy mode, the device ID will be different.
```

Comment:
I feel we should but this into a Note.

Reply:
Delete this paragraph and add a link to cookie management at the end of #3:
3. Device id in cookie storage. Refer to [cookies management](./#cookie-management) for more details